graylog2-server/src/test/java/org/graylog/plugins/views/search/elasticsearch/searchtypes/ESMessageListTest.java

```diff
@@ -88,6 +92,47 @@ public void appliesDecoratorsToQueryStringIfHighlightingActivated() {
         JsonPathAssert.assertThat(doc).jsonPathAsString("$.highlight.highlight_query.query_string.query").isEqualTo("Foobar!");
     }
 
+    @Test
+    public void passesTypeOfSortingFieldAsUnmappedType() {
+        final MessageList messageList = someMessageListWithSorting();
+        final ESGeneratedQueryContext context = mockQueryContext(messageList);
+        when(context.fieldTypes(Collections.singleton("stream1"))).thenReturn(Collections.singletonMap("somefield", Collections.singleton("long")));
+
+        final ESGeneratedQueryContext queryContext = generateQueryPartWithContextFor(messageList, true, Collections.emptySet(), context);
+
+        final DocumentContext doc = JsonPath.parse(queryContext.searchSourceBuilder(messageList).toString());
+        JsonPathAssert.assertThat(doc).jsonPathAsString("$.sort[0].somefield.unmapped_type").isEqualTo("long");
+    }
+
+    @Test
```

Comment:
This and the next test would become obsolete, if you followed my suggestion above to move that logic into `FieldTypeLookup`. I would add a few simpler unit tests for `FieldTypesLookup` instead.

Reply:
✅